tracing-subscriber/src/fmt/format.rs

```diff
@@ -280,8 +309,61 @@ where
     }
 }
 
-// === impl FormatFields ===
+#[cfg(feature = "json")]
+impl<N, T> FormatEvent<N> for Format<Json, T>
+where
+    N: for<'writer> FormatFields<'writer>,
+    T: FormatTime,
+{
+    fn format_event(
+        &self,
+        ctx: &span::Context<'_, N>,
+        writer: &mut dyn fmt::Write,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        let mut timestamp = String::new();
+        self.timer.format_time(&mut timestamp)?;
+
+        #[cfg(feature = "tracing-log")]
+        let normalized_meta = event.normalized_metadata();
+        #[cfg(feature = "tracing-log")]
+        let meta = normalized_meta.as_ref().unwrap_or_else(|| event.metadata());
+        #[cfg(not(feature = "tracing-log"))]
+        let meta = event.metadata();
+
+        let mut serializer = Serializer::new(WriteAdaptor::new(writer));
+
+        let mut visit = || {
+            let mut serializer = serializer.serialize_map(None)?;
+
+            serializer.serialize_key("timestamp")?;
+            serializer.serialize_value(timestamp.trim())?;
+            serializer.serialize_key("level")?;
+            serializer.serialize_value(&format!("{}", meta.level()))?;
```

Comment:
Could we implement `AsRef<str>` for `Level` so we'd save one allocation here?

Reply:
I'm speaking from a position of ignorance, but would the [`SerializeLevel`](https://tracing.rs/tracing_serde/struct.serializelevel) struct in `tracing-serde` do the trick (alongside all the other [structs in tracing-serde](https://tracing.rs/tracing_serde/index.html#structs))?

Reply:
Yeah. And I'm speaking from a position of not knowing the codebase good enough yet. Going with the `tracing-serde` structs.

Reply:
No worries! I just wasn't sure _how_ much of `tracing` you were familiar with, so I err'd on the side of caution.

Reply:
In this way maybe... Now we can't use ANSI at all of enabling `json` feature. https://github.com/pimeys/tracing/commit/3e4c8d9a10862b8770e2dc0f36b5bb1d46d0e394

Reply:
I can't even say it's my first day anymore, because this is my second day with `tracing` codebase...

Reply:
What you think is best, revert https://github.com/pimeys/tracing/commit/6912d62e00a1fc09eee9795a8fe17dba9e354e03

Reply:
I'm sorry about that back and forth—feel free to ping me on discord in `tracing-users` if you need anything: https://discord.gg/EeF3cQw